### PR TITLE
HHH-16935 Fix inconsistent method name

### PR DESCRIPTION
**Pull request by quaff**
https://hibernate.atlassian.net/browse/HHH-16935

### CODE REVIEW

**Comment by quaff** [comment]
@sebersole Please review.

**Comment by quaff** [comment]
> That looks reasonable, thanks! I'll let @sebersole decide about the final method name, but could you re-introduce the old method name as a deprecated delegate please? Otherwise I'm not sure which platforms we might break, and we wouldn't want to do that in a micro.

@Sanne Updated by introducing deprecated default method as your suggestion.

**Comment by quaff** [comment]
It's not just renaming but also logic error fix, please check it ASAP. @Sanne @sebersole 
https://github.com/hibernate/hibernate-orm/blob/0897a90a9e674d1a56633622fcd6a240d7dbda41/hibernate-core/src/main/java/org/hibernate/boot/internal/MetadataBuilderImpl.java#L973-L976

File: hibernate-core/src/main/java/org/hibernate/boot/spi/MetadataBuildingOptions.java [hunk 1]
```diff
@@ -243,5 +243,15 @@ default boolean isXmlMappingEnabled() {
 	/**
 	 * Check to see if extensions can be hosted in CDI
 	 */
-	boolean disallowExtensionsInCdi();
+	boolean isAllowExtensionsInCdi();
```
Comment (gavinking) [inline hibernate-core/src/main/java/org/hibernate/boot/spi/MetadataBuildingOptions.java:246]: This is an extremely awkward-sounding name. 

How about something like `isCdiIntegrationEnabled()`. WDYT, @sebersole?

**Review by gavinking — CHANGES_REQUESTED** [review]

File: hibernate-core/src/main/java/org/hibernate/boot/spi/MetadataBuildingOptions.java (hunk 1 above)
Comment (quaff) [reply hibernate-core/src/main/java/org/hibernate/boot/spi/MetadataBuildingOptions.java:246]: > This is an extremely awkward-sounding name.
> 
> How about something like `isCdiIntegrationEnabled()`. WDYT, @sebersole?

Please keep in mind `AvailableSettings.ALLOW_EXTENSIONS_IN_CDI` is public, consistency or compatibility will be broken if renaming is made again.
https://github.com/hibernate/hibernate-orm/blob/0897a90a9e674d1a56633622fcd6a240d7dbda41/hibernate-core/src/main/java/org/hibernate/cfg/AvailableSettings.java#L1356

**Comment by quaff** [comment]
What's the decision?

**Comment by sebersole** [comment]
Sorry, missed the notifications from this.  :+1:   Jan is going to rebase this and apply

**Comment by jrenaat** [comment]
> Sorry, missed the notifications from this. 👍 Jan is going to rebase this and apply

Done.

**Comment by quaff** [comment]
It should be backported to 6.2.x and 6.3.x.

**Comment by mgvinuesa** [comment]
Hello, I think this change it is related with this bug

https://hibernate.atlassian.net/jira/software/c/projects/HHH/issues/HHH-16881?filter=allissues

I don't know if you have time to check it, but currently it is not possible to use RevisionListener with Spring Boot 3.1.X

Thanks in advance

**Comment by quaff** [comment]
> Hello, I think this change it is related with this bug
> 
> https://hibernate.atlassian.net/jira/software/c/projects/HHH/issues/HHH-16881?filter=allissues
> 
> I don't know if you have time to check it, but currently it is not possible to use RevisionListener with Spring Boot 3.1.X
> 
> Thanks in advance

I missed checking `Helper.allowExtensionsInCdi()`.

**Comment by mgvinuesa** [comment]
I created a PR yesterday 

https://github.com/hibernate/hibernate-orm/pull/7446

**Pull request merged**